fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -499,8 +499,13 @@ where
         self.input_contracts.check(&contract_id)?;
 
         // Fetch the storage contract
-        let contract = super::contract::contract(self.storage, &contract_id)?;
-        let contract = contract.as_ref().as_ref();
+        // If $rC is greater than the code size, zero bytes are filled in.
+        let contract = super::contract::contract_subsection(
+            self.storage,
+            &contract_id,
+            length_unpadded as usize,
+        )?;
```

Comment:
Our specification says:

> Copy `$rC` bytes of code starting at `$rB` for contract with ID equal to the 32 bytes in memory starting at `$rA` into memory starting at `$ssp`.

You are missing `$rb` offset. We need to copy the subset starting from `[$rb, $rb + $rc]`.

Reply:
I fixed this, but I still need to add test to check it is working.

Reply:
K. Added test.